tasks/clear-cache-reference-data.test.js

```diff
@@ -0,0 +1,118 @@
+const proxyquire = require('proxyquire').noPreserveCache()
+
+const redisClient = {}
+
+const { clearCacheReferenceData, main } = proxyquire(
+  './clear-cache-reference-data',
+  {
+    '../config/redis-store.js': () => {
+      return {
+        client: Promise.resolve(redisClient),
+      }
+    },
+  }
+)
+
+describe('Clear cache reference data', function () {
+  beforeEach(function () {
+    redisClient.scanAsyncData = []
+    redisClient.scanAsync = sinon.stub().returns({
+      then: f =>
+        f(
+          redisClient.scanAsyncData.shift() || [
+            '0',
+            [
+              'cache:v2:GET./api/reference/locations/1',
+              'cache:v2:GET./api/reference/locations/2',
+            ],
+          ]
+        ),
+    })
+    redisClient.del = sinon.stub().resolves()
+  })
+
+  context('when ran via the command line', function () {
+    beforeEach(function () {
+      sinon.stub(process.stderr, 'write')
+      sinon.stub(process.stdout, 'write')
+      sinon.stub(process, 'exit')
+    })
+
+    afterEach(function () {
+      process.stderr.write.restore()
+      process.stdout.write.restore()
+      process.exit.restore()
+    })
+
+    context('when no arg is passed', function () {
+      beforeEach(function () {
+        process.argv[2] = undefined
+      })
+
+      it('tells the user that it needs an arg and exits with code 1', async function () {
+        await main()
+
+        expect(process.stderr.write).to.have.been.calledOnceWithExactly(
+          'Missing required arg `resourceName`.\n'
+        )
+        expect(process.exit).to.have.been.calledOnceWithExactly(1)
+      })
+    })
+
+    context('when an arg is passed', function () {
+      beforeEach(function () {
+        process.argv[2] = 'locations'
+      })
+
+      it('logs some useful information and exits with code 0', async function () {
+        await main()
+
+        expect(process.stdout.write).to.have.been.called
+        expect(process.exit).to.have.been.calledOnceWithExactly(0)
+      })
+    })
+  })
+
+  context('when called without an arg', function () {
```

Comment:
Is this different from the context above `when no arg is passed`?

Reply:
The contexts above are in `when ran via the command line` and call `main` as to emulate the script being ran from the command line, instead of directly calling `clearCacheReferenceData`